shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -175,18 +175,26 @@ - (void)ensureSemanticsEnabled {
 
 - (void)setViewController:(FlutterViewController*)viewController {
   FML_DCHECK(self.iosPlatformView);
-  _viewController = [viewController getWeakPtr];
+  _viewController = viewController ? [viewController getWeakPtr] : fml::WeakPtr<FlutterViewController>();
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
 
-  __block FlutterEngine* blockSelf = self;
-  self.flutterViewControllerWillDeallocObserver =
-      [[NSNotificationCenter defaultCenter] addObserverForName:FlutterViewControllerWillDealloc
-                                                        object:viewController
-                                                         queue:[NSOperationQueue mainQueue]
-                                                    usingBlock:^(NSNotification* note) {
-                                                      [blockSelf notifyViewControllerDeallocated];
-                                                    }];
+  if (_flutterViewControllerWillDeallocObserver) {
```

Comment:
This LGTM too. @gaaclarke to double check.

Reply:
I'd prefer:
```objc
if (viewController) {
  __block FlutterEngine* blockSelf = self;
  self.flutterViewControllerWillDeallocObserver =
      [[NSNotificationCenter defaultCenter] addObserverForName:FlutterViewControllerWillDealloc
                                                        object:viewController
                                                         queue:[NSOperationQueue mainQueue]
                                                    usingBlock:^(NSNotification* note) {
                                                      [blockSelf notifyViewControllerDeallocated];
                                                    }];
} else {
  self.flutterViewControllerWillDeallocObserver = nil;
}
```

You aren't taking advantage of the setter that we've made in this class.

Reply:
you are right, I forgot about the setter. will change

Reply:
updated